dialogue-annotations-processor/src/test/resources/com/palantir/myservice/service/MyServiceDialogueServiceFactory.java.generated

```diff
@@ -156,6 +174,18 @@ public final class MyServiceDialogueServiceFactory implements DialogueServiceFac
                             paramsMaybeCustomOptionalHeader3ValueEncoder.toParamValues(
                                     maybeCustomOptionalHeader3Value.get()));
                 }
+                _request.putAllHeaderParams(
+                        "Custom-Header1",
+                        customListHeader.stream()
+                                .map(_parameterSerializer::serializeString)
```

Comment:
Debated adding some shared code for this, but generally in favour of generating as much as we can, as that's compile-time validation and breakable, as opposed to library code.